File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/stringee/receiver/CallActionOwnership.java
package com.stringee.apptoappcallsample.stringee.receiver;

/** Pure ownership rule for targeting an active incoming session from a notification action. */
final class CallActionOwnership {
    private CallActionOwnership() {
    }

    static boolean canTargetActiveSession(boolean hasMatchingIncomingSession,
                                          boolean ownsPush,
                                          long requestedSessionGeneration,
                                          long currentSessionGeneration) {
        return hasMatchingIncomingSession
                && ownsPush
                && (requestedSessionGeneration < 0
                || requestedSessionGeneration == currentSessionGeneration);
    }
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/stringee/receiver/CallActionOwnership.kt
package com.stringee.kotlin_onetoonecallsample.stringee.receiver

/** Pure ownership rule for targeting an active incoming session from a notification action. */
internal object CallActionOwnership {
    @JvmStatic fun canTargetActiveSession(
        hasMatchingIncomingSession: Boolean,
        ownsPush: Boolean,
        requestedSessionGeneration: Long,
        currentSessionGeneration: Long
    ): Boolean {
        return hasMatchingIncomingSession
                && ownsPush
                && (requestedSessionGeneration < 0
                || requestedSessionGeneration == currentSessionGeneration)
    }
}
